### PR TITLE
[Bridge] [Doctrine] [Validator] Added support \IteratorAggregate for UniqueEntityValidator

### PR DESCRIPTION
**Pull request by Disparity**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Expand the list of supported types of results returned from the repositories.
Added processing of type \IteratorAggregate (and as a consequence doctrine Collection)

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Can you please add a test case?

**Comment by Disparity** [comment]
Added tests to check the functionality different types of "Collections".

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by xabbuh** [comment]
Is this supposed to be a bugfix or a new feature?

**Comment by Disparity** [comment]
@xabbuh, these changes are closer to a new feature than to bug fixes.
This allows the validator working with traversable results (which include doctrine collection), and not only with iterators and arrays.

File: src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php [hunk 1]
```diff
@@ -137,6 +138,17 @@ private function createSchema(ObjectManager $em)
...
+    public function resultTypesProvider()
+    {
+        $entity = new SingleIntIdEntity(1, 'foo');
+
+        return array(
+            array($entity, array($entity)),
+            array($entity, new \ArrayIterator(array($entity))),
+            array($entity, new ArrayCollection(array($entity))),
+        );
+    }
```
Comment (xabbuh) [inline src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php]: Imo for better readability the data provider should be near (above or below) the test method.

Comment (Disparity) [inline src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php]: I looked existing code - not so simple, there are different tests in different approaches. But I agree that the data provider close to the test easier to read. Fixed.

**Comment by fabpot** [comment]
:+1: I would consider this a bugfix rather than a new feature.

**Comment by xabbuh** [comment]
:+1: for merging this into Symfony 2.3

**Comment by fabpot** [comment]
Thank you @Disparity.

**Pull request closed without merging**